### DISCUSSION TITLE
Error when launching Rust debug

### DISCUSSION THREAD

**Discussion by fnuttens** [helix-editor/helix#5654] [Q&A]
Hi! Since I'm not sure whether I stumbled on a bug or I'm simply doing things wrong, I'd rather open this discussion than an issue 😅

I'm trying to debug a dummy console app on Windows, made with `cargo new`.

Needless to say that the program compiles et executes with no hiccup.  
However, I get the following error when I run the command `:debug-start binary target\debug\playground.exe`:

```
error: 'Content-Length' is not a valid command.
*a few moments later*
Failed to initialize debug adapter: request timed out
```

After that the interface gets all buggy, I have to restart helix.

The logs look like this:

```log
2023-01-23T22:57:58.157 helix_dap::transport [INFO] -> DAP {"type":"request","seq":0,"command":"initialize","arguments":{"adapterID":"lldb-vscode","clientID":"hx","clientName":"helix","columnsStartAt1":true,"linesStartAt1":true,"locale":"en-us","pathFormat":"path","supportsInvalidatedEvent":false,"supportsMemoryReferences":false,"supportsProgressReporting":false,"supportsRunInTerminalRequest":true,"supportsVariablePaging":false,"supportsVariableType":true}}
2023-01-23T22:58:00.868 helix_dap::transport [ERROR] err: <- StreamClosed
2023-01-23T22:58:00.868 helix_lsp::transport [ERROR] err: <- StreamClosed
```

The time out might indicate an issue with the rust debugger itself, but `hx --health rust` is all green 🤷🏻‍♂️ 

```
Configured language server: rust-analyzer
Binary for language server: C:\Users\fnuttens\.cargo\bin\rust-analyzer.exe
Configured debug adapter: lldb-vscode
Binary for debug adapter: C:\Users\fnuttens\.vscode\extensions\vadimcn.vscode-lldb-1.8.1\lldb\bin\lldb-vscode.exe
Highlight queries: ✓
Textobject queries: ✓
Indent queries: ✓
```

Any idea what I might be doing wrong?

**Comment by quantonganh** [comment]
Instead of using `:debug-start`, have you tried:

- `<space>`, then `g` to go to debug
- `b` to toggle breakpoint
- `l` to launch debug target
- enter the binary path

Don't forget the check [the latest wiki](https://github.com/helix-editor/helix/wiki/Debugger-Configurations#rust).

**Reply by fnuttens** [reply]
Thanks for helping!

Yes I've tried this and I had the same error… Maybe I shouldn't be using the MSVC toolchain?  
I will also try to use `codelldb` instead of `lldb-vscode`.

**Reply by quantonganh** [reply]
```
Binary for debug adapter: C:\Users\fnuttens\.vscode\extensions\vadimcn.vscode-lldb-1.8.1\lldb\bin\lldb-vscode.exe
```

Looks like you are using `lldb-vscode` from [codelldb](https://github.com/vadimcn/codelldb).

Have you tried to install `llvm` by using [chocolatey](https://community.chocolatey.org/packages/llvm)?

**Comment by fnuttens** [comment]
Hey @quantonganh, sorry for not replying sooner, I just gave this another try and I finally got the hang of it! In the end I managed to use codelldb with the configuration from the wiki. Thanks!
